trackpy/linking/subnetlinker.py

```diff
@@ -427,24 +427,41 @@ def subnet_linker_nonrecursive(source_set, dest_set, search_range, **kwargs):
 
     return sn_spl, sn_dpl
 
-
-def subnet_linker_numba(source_set, dest_set, search_range, **kwargs):
-    if len(source_set) == 0 and len(dest_set) == 1:
+def subnet_linker_numba(source_set, dest_set, search_range,
+                        hybrid=True, **kwargs):
+    """Link a subnet using a numba-accelerated algorithm.
+
+    Since this is meant to be the highest-performance option, it
+    has some special behaviors:
+
+    - Each source particle's forward_cands must be sorted by distance.
+    - If the 'hybrid' option is true, subnets with only 1 source or
+      destination particle, or with at most 4 source particles and
+      4 destination particles, are solved using the recursive
+      pure-Python algorithm, which has much less overhead since
+      it does not convert to a numpy representation.
+    """
+    lss = len(source_set)
+    lds = len(dest_set)
+    if lss == 0 and lds == 1:
         # no backwards candidates: particle will get a new track
         return [None], [dest_set.pop()]
-    elif len(source_set) == 1 and len(dest_set) == 1:
+    elif lss == 1 and lds == 1:
         # one backwards candidate and one forward candidate
         return [source_set.pop()], [dest_set.pop()]
-    elif len(source_set) == 1 and len(dest_set) == 0:
+    elif lss == 1 and lds == 0:
         # particle is lost. Not possible with default Linker implementation.
         return [source_set.pop()], [None]
 
-    # sort candidates and add in penalty for not linking
+    # Forward candidates were already sorted by Linker.assign_links()
     for _s in source_set:
-        _s.forward_cands.sort(key=lambda x: x[1])
         _s.forward_cands.append((None, search_range))
```

Comment:
Not sure if adding the null candidate should be done in the subnetlinker or before the subnetlinker. It seems cleaner to me to do it before? Are there any reasons for doing it like this?